src/codegen/impl_partialeq.rs

```diff
@@ -14,16 +14,12 @@ pub fn gen_partialeq_impl(
 ) -> Option<proc_macro2::TokenStream> {
     let mut tokens = vec![];
```

Comment:
You should just `debug_assert!(!item.is_opaque(..))` if we decide not to derive partialeq for these, then remove the conditional below.